core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1138,115 @@ def delete_playthroughs_multi(cls, playthrough_ids):
         cls.delete_multi(instances)
 
 
+class LearnerAnswerDetailsModel(base_models.BaseModel):
+    """Model for storing the answer details, learner enters when they
+    are asked for it.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # ID of the entity.
+    entity_id = ndb.StringProperty(required=True, indexed=True)
+    # The type of entity, that whether it is state, question, or any other
+    # object if it is added in future.
+    entity_type = ndb.StringProperty(required=True, indexed=True)
+    # The answer submitted by the learner.
+    learner_answer = ndb.StringProperty(required=True, indexed=False)
+    # The response submitted by the learner.
+    learner_response = ndb.StringProperty(required=True, indexed=False)
+    # The time at which the response was created.
+    created_on = ndb.DateTimeProperty(required=True, indexed=False)
+    # Whether the response received has been resolved.
+    is_resolved = ndb.BooleanProperty(default=False, required=True)
+
+    @classmethod
+    def get_entity_id(cls, exploration_id, state_name):
```

Comment:
This function is only for an exploration, right? For any other entities, it would be different. So, maybe rename it to `get_exploration_entity_id` or something similar?

Reply:
Yeah it is for exploration only.